WordPress/src/main/java/org/wordpress/android/ui/prefs/notifications/NotificationsSettingsFragment.java

```diff
@@ -554,29 +555,52 @@ private void updateUIForNotificationsEnabledState() {
                 category.removePreference(category.getPreference(TYPE_COUNT));
             } else if (!mNotificationsEnabled && category.getPreferenceCount() == TYPE_COUNT) {
                 Preference disabledMessage = new Preference(getActivity());
-                disabledMessage.setSummary(R.string.notifications_disabled);
-                disabledMessage.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
-                    @Override
-                    public boolean onPreferenceClick(Preference preference) {
-                        Intent intent = new Intent();
-                        intent.setAction(Settings.ACTION_APPLICATION_DETAILS_SETTINGS);
-                        Uri uri =
-                                Uri.fromParts("package", getActivity().getApplicationContext().getPackageName(), null);
-                        intent.setData(uri);
-
-                        startActivity(intent);
-                        return true;
-                    }
-                });
-
                 category.addPreference(disabledMessage);
             }
 
             if (category.getPreferenceCount() >= TYPE_COUNT
                 && category.getPreference(TYPE_COUNT - 1) != null) {
                 category.getPreference(TYPE_COUNT - 1).setEnabled(mNotificationsEnabled);
             }
+
+            if (category.getPreferenceCount() > TYPE_COUNT
+                && category.getPreference(TYPE_COUNT) != null) {
+                updateDisabledMessagePreference(category.getPreference(TYPE_COUNT));
+            }
+        }
+    }
+
+    private void updateDisabledMessagePreference(Preference disabledMessagePreference) {
+        boolean isAlwaysDenied =
```

Comment:
Would it read better if isAlwaysDenied and summaryResId are moved out of this method, and could also reuse it on line 585 as 578 is also same.

Reply:
Good idea!
Done: 292ed7cc1c4aef6a48c10b89cbe4f04d6993c2ab